### PR TITLE
Fix Azure.Core when running on Blazor WASM

### LINKED ISSUE #13298: [BUG] BlobClient.DownloadAsync() Response x-ms-client-request-id  does not match the original expected request id
Using the latest Blazor WebAssembly sdk I am trying to download a Blob's content using BlobClient.DownloadAsync()

**Expected behavior**
I expect the blob to download and on the first attempt if its been left a while it does appear to work but then never again after.

**Actual behavior (include Exception or Stack Trace)**
Upon downloading of the blob an exception occurs:

Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Response x-ms-client-request-id '<GUID1>' does not match the original expected request id, '<GUID2>'.

This is the output in the browser console:
```
 Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Response x-ms-client-request-id '82d5f0bb-bd83-4ef5-8f3a-fa5165f4737a' does not match the original expected request id, 'fcacef5f-0472-44c7-b36d-bafdf2768451'.
      Status: 200 (OK)
      
      Headers:
      Accept-Ranges: bytes
      Date: Mon, 06 Jul 2020 19:18:05 GMT
      ETag: "0x8D82021DE355DD6"
      Server: Windows-Azure-Blob/1.0,Microsoft-HTTPAPI/2.0
      x-ms-blob-type: BlockBlob
      x-ms-client-request-id: 82d5f0bb-bd83-4ef5-8f3a-fa5165f4737a
      x-ms-creation-time: Sat, 04 Jul 2020 13:55:10 GMT
      x-ms-lease-state: available
      x-ms-lease-status: unlocked
      x-ms-request-id: e415a9cd-001e-000b-7dca-53dfb7000000
      x-ms-server-encrypted: true
      x-ms-version: 2019-07-07
      Content-Length: 1284435
      Content-MD5: uQBQz3kRBFQgZPSAcV27LA==
      Content-Type: application/json
      Last-Modified: Sat, 04 Jul 2020 13:55:10 GMT
      
Azure.RequestFailedException: Response x-ms-client-request-id '82d5f0bb-bd83-4ef5-8f3a-fa5165f4737a' does not match the original expected request id, 'fcacef5f-0472-44c7-b36d-bafdf2768451'.
Status: 200 (OK)

Headers:
Accept-Ranges: bytes
Date: Mon, 06 Jul 2020 19:18:05 GMT
ETag: "0x8D82021DE355DD6"
Server: Windows-Azure-Blob/1.0,Microsoft-HTTPAPI/2.0
x-ms-blob-type: BlockBlob
x-ms-client-request-id: 82d5f0bb-bd83-4ef5-8f3a-fa5165f4737a
x-ms-creation-time: Sat, 04 Jul 2020 13:55:10 GMT
x-ms-lease-state: available
x-ms-lease-status: unlocked
x-ms-request-id: e415a9cd-001e-000b-7dca-53dfb7000000
x-ms-server-encrypted: true
x-ms-version: 2019-07-07
Content-Length: 1284435
Content-MD5: uQBQz3kRBFQgZPSAcV27LA==
Content-Type: application/json
Last-Modified: Sat, 04 Jul 2020 13:55:10 GMT

  at Azure.Storage.StorageRequestValidationPipelinePolicy.OnReceivedResponse (Azure.Core.HttpMessage message) <0x34b6270 + 0x000d2> in <filename unknown>:0 
  at Azure.Core.Pipeline.HttpPipelineSynchronousPolicy.ProcessAsync (Azure.Core.HttpMessage message, System.ReadOnlyMemory`1[T] pipeline) <0x33c0748 + 0x00114> in <filename unknown>:0 
  at Azure.Core.Pipeline.RetryPolicy.ProcessAsync (Azure.Core.HttpMessage message, System.ReadOnlyMemory`1[T] pipeline, System.Boolean async) <0x33a8080 + 0x0012c> in <filename unknown>:0 
  at Azure.Core.Pipeline.RetryPolicy.ProcessAsync (Azure.Core.HttpMessage message, System.ReadOnlyMemory`1[T] pipeline, System.Boolean async) <0x33a8080 + 0x00270> in <filename unknown>:0 
  at Azure.Core.Pipeline.HttpPipelineSynchronousPolicy.ProcessAsync (Azure.Core.HttpMessage message, System.ReadOnlyMemory`1[T] pipeline) <0x33c0748 + 0x00100> in <filename unknown>:0 
  at Azure.Core.Pipeline.HttpPipelineSynchronousPolicy.ProcessAsync (Azure.Core.HttpMessage message, System.ReadOnlyMemory`1[T] pipeline) <0x33c0748 + 0x00100> in <filename unknown>:0 
  at Azure.Core.Pipeline.HttpPipelineSynchronousPolicy.ProcessAsync (Azure.Core.HttpMessage message, System.ReadOnlyMemory`1[T] pipeline) <0x33c0748 + 0x00100> in <filename unknown>:0 
  at Azure.Storage.Blobs.BlobRestClient+Blob.DownloadAsync (Azure.Core.Pipeline.ClientDiagnostics clientDiagnostics, Azure.Core.Pipeline.HttpPipeline pipeline, System.Uri resourceUri, System.String version, System.String snapshot, System.Nullable`1[T] timeout, System.String range, System.String leaseId, System.Nullable`1[T] rangeGetContentHash, System.Nullable`1[T] rangeGetContentCRC64, System.String encryptionKey, System.String encryptionKeySha256, System.Nullable`1[T] encryptionAlgorithm, System.Nullable`1[T] ifModifiedSince, System.Nullable`1[T] ifUnmodifiedSince, System.Nullable`1[T] ifMatch, System.Nullable`1[T] ifNoneMatch, System.String requestId, System.Boolean async, System.String operationName, System.Threading.CancellationToken cancellationToken) <0x35b43e0 + 0x002c6> in <filename unknown>:0 
  at System.Threading.Tasks.ValueTask`1[TResult].get_Result () <0x35cfde8 + 0x0003c> in <filename unknown>:0 
  at Azure.Storage.Blobs.Specialized.BlobBaseClient.StartDownloadAsync (Azure.HttpRange range, Azure.Storage.Blobs.Models.BlobRequestConditions conditions, System.Boolean rangeGetContentHash, System.Int64 startOffset, System.Boolean async, System.Threading.CancellationToken cancellationToken) <0x35b2e50 + 0x004be> in <filename unknown>:0 
  at Azure.Storage.Blobs.Specialized.BlobBaseClient.DownloadInternal (Azure.HttpRange range, Azure.Storage.Blobs.Models.BlobRequestConditions conditions, System.Boolean rangeGetContentHash, System.Boolean async, System.Threading.CancellationToken cancellationToken) <0x35aaeb0 + 0x002b8> in <filename unknown>:0 
  at Azure.Storage.Blobs.Specialized.BlobBaseClient.DownloadAsync (Azure.HttpRange range, Azure.Storage.Blobs.Models.BlobRequestConditions conditions, System.Boolean rangeGetContentHash, System.Threading.CancellationToken cancellationToken) <0x35a2830 + 0x000fa> in <filename unknown>:0 
  at Azure.Storage.Blobs.Specialized.BlobBaseClient.DownloadAsync (System.Threading.CancellationToken cancellationToken) <0x35a20c8 + 0x000fc> in <filename unknown>:0 
  at Azure.Storage.Blobs.Specialized.BlobBaseClient.DownloadAsync () <0x359d2c8 + 0x000ea> in <filename unknown>:0 
  at Blazor.Shared.ViewReportBase.OnInitializedAsync () [0x000a6] in C:\Users\Anthony\Source\Repos\AccessibilityApp\src\AccessibilityReportViewer\Blazor.Shared\ViewReport.razor.cs:34 
  at Microsoft.AspNetCore.Components.ComponentBase.RunInitAndSetParametersAsync () <0x2d12630 + 0x0013a> in <filename unknown>:0 
  at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask (System.Threading.Tasks.Task taskToHandle) <0x2fdc928 + 0x000b6> in <filename unknown>:0 

```

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Full repo can be found at https://github.com/AnthonyDewhirst/BlazorADandBlobs
2. Change the details in appsettings.json and run. 
3. Login to your selected AD tennant
4. Select Reports
5. Select to view a report

**Environment:**
- MS Edge Version 83.0.478.58 (Official build) (64-bit)
-  <TargetFramework>netstandard2.1</TargetFramework>
    <RazorLangVersion>3.0</RazorLangVersion>

 -  <PackageReference Include="Azure.Identity" Version="1.1.1" />
    <PackageReference Include="Azure.Storage.Blobs" Version="12.4.4" />
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="3.2.0" />
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.Build" Version="3.2.0" PrivateAssets="all" />
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="3.2.0" PrivateAssets="all" />
    <PackageReference Include="Microsoft.Authentication.WebAssembly.Msal" Version="3.2.0" />
    <PackageReference Include="System.Net.Http.Json" Version="3.2.1" />
 - .NET Core SDK (reflecting any global.json):
 Version:   3.1.301
 Commit:    7feb845744

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.301\

Host (useful for support):
  Version: 3.1.5
  Commit:  65cd789777

.NET Core SDKs installed:
  1.0.0-rc4-004771 [C:\Program Files\dotnet\sdk]
  1.0.0 [C:\Program Files\dotnet\sdk]
  1.0.2 [C:\Program Files\dotnet\sdk]
  1.0.3 [C:\Program Files\dotnet\sdk]
  1.0.4 [C:\Program Files\dotnet\sdk]
  2.1.4 [C:\Program Files\dotnet\sdk]
  2.1.103 [C:\Program Files\dotnet\sdk]
  2.1.104 [C:\Program Files\dotnet\sdk]
  2.1.200 [C:\Program Files\dotnet\sdk]
  2.1.201 [C:\Program Files\dotnet\sdk]
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.302 [C:\Program Files\dotnet\sdk]
  2.1.400 [C:\Program Files\dotnet\sdk]
  2.1.401 [C:\Program Files\dotnet\sdk]
  2.1.402 [C:\Program Files\dotnet\sdk]
  2.1.403 [C:\Program Files\dotnet\sdk]
  2.1.500 [C:\Program Files\dotnet\sdk]
  2.1.502 [C:\Program Files\dotnet\sdk]
  2.1.503 [C:\Program Files\dotnet\sdk]
  2.1.504 [C:\Program Files\dotnet\sdk]
  2.1.505 [C:\Program Files\dotnet\sdk]
  2.1.507 [C:\Program Files\dotnet\sdk]
  2.1.509 [C:\Program Files\dotnet\sdk]
  2.1.511 [C:\Program Files\dotnet\sdk]
  2.1.600 [C:\Program Files\dotnet\sdk]
  2.1.601 [C:\Program Files\dotnet\sdk]
  2.1.602 [C:\Program Files\dotnet\sdk]
  2.1.800-preview-009677 [C:\Program Files\dotnet\sdk]
  2.2.200-preview-009748 [C:\Program Files\dotnet\sdk]
  2.2.200 [C:\Program Files\dotnet\sdk]
  2.2.201 [C:\Program Files\dotnet\sdk]
  2.2.202 [C:\Program Files\dotnet\sdk]
  2.2.400-preview-010195 [C:\Program Files\dotnet\sdk]
  3.1.301 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.19 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
 - Visual Studio 16.6.3

### LINKED ISSUE #20125: [BUG] HttpClientTransport calls CreateDefaultClient regardless of argument / not compatible with Blazor
**Describe the bug**
Cannot instantiate HttpClientTransport using HttpClient, it calls CreateDefaultClient() regardless of the argument given. This happens with any Client SDK or with an autorest generated one.
Is the SDK not supposed to be compatible with Blazor WASM?

**Expected behavior**
What is the expected behavior?
Instantiate a BlobClient and/or HttpClientTransport based on HttpClient that is compatible with Blazor WASM

**Actual behavior (include Exception or Stack Trace)**
What is the actual behavior?
```
crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: The type initializer for 'Azure.Core.Pipeline.HttpClientTransport' threw an exception.
System.TypeInitializationException: The type initializer for 'Azure.Core.Pipeline.HttpClientTransport' threw an exception.
 ---> System.PlatformNotSupportedException: Operation is not supported on this platform.
   at System.Net.Http.SocketsHttpHandler.get_MaxConnectionsPerServer()
   at Azure.Core.Pipeline.ServicePointHelpers.SetLimits(SocketsHttpHandler socketsHttpHandler)
   at Azure.Core.Pipeline.HttpClientTransport.CreateDefaultClient()
   at Azure.Core.Pipeline.HttpClientTransport..ctor()
   at Azure.Core.Pipeline.HttpClientTransport..cctor()
   --- End of inner exception stack trace ---
   at QuoteBux.Pages.Index.test() in C:\Users\mathi\source\repos\QuoteBux\Pages\Index.razor:line 25
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle)
```
**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Create a new Blazor WASM project
2. run this 
```
  var httpClientTransport = new HttpClientTransport(HttpClient);
        var options = new BlobClientOptions() {Transport = httpClientTransport};
        var client = new BlobClient(new Uri(connectionString), options);
```

**Environment:**
 - Name and version of the Library package used: Azure.Storage.Blobs 12.8.1
 - Hosting platform or OS and .NET runtime version 
 ```
.NET SDK (reflecting any global.json):
 Version:   5.0.201
 Commit:    a09bd5c86c

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.201\

Host (useful for support):
  Version: 5.0.4
  Commit:  f27d337295

.NET SDKs installed:
  3.1.402 [C:\Program Files\dotnet\sdk]
  5.0.103 [C:\Program Files\dotnet\sdk]
  5.0.201 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```
 - IDE and version : [e.g. Visual Studio 16.3] VS 16.9

### PR DESCRIPTION
**Pull request by pakrym**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/20125

Fixes https://github.com/Azure/azure-sdk-for-net/issues/13298

### CODE REVIEW

File: sdk/core/Azure.Core/src/Azure.Core.csproj [hunk 1]
```diff
@@ -13,6 +13,10 @@
     <EnableClientSdkAnalyzers>true</EnableClientSdkAnalyzers>
   </PropertyGroup>
 
+  <ItemGroup>
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/Azure.Core.csproj:16]: Enables the compatibility analyzer.

File: sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs [hunk 2]
```diff
@@ -147,6 +143,46 @@ private static HttpClient CreateDefaultClient()
             };
         }
 
+        private static HttpMessageHandler CreateDefaultHandler()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER")))
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:148]: This is wordy  but is the syntax that analyzer understands.

File: sdk/core/Azure.Core/tests/HttpClientTransportFunctionalTest.cs [hunk 3]
```diff
@@ -33,5 +36,31 @@ protected override HttpPipelineTransport GetTransport(bool https = false)
...
+        public void UsesHttpClientTransportOnNetStandart()
+        {
+            var transport = (HttpClientTransport)GetTransport();
+            HttpClient transportClient = transport.Client;
+            var handler = (HttpClientHandler)GetHandler(transportClient);
+            Assert.AreEqual(50, handler.MaxConnectionsPerServer);
+        }
+#endif
+
+        private static object GetHandler(HttpClient transportClient)
```
Comment (pakrym) [inline sdk/core/Azure.Core/tests/HttpClientTransportFunctionalTest.cs:61]: No a fan of this myself but want to make sure nothing is broken during refactorings.

File: sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs [hunk 4]
```diff
@@ -83,13 +85,16 @@ private async ValueTask ProcessAsync(HttpMessage message, bool async)
 #if NET5_0
                 if (!async)
                 {
-                    responseMessage = _client.Send(httpRequest, HttpCompletionOption.ResponseHeadersRead, message.CancellationToken);
+                    // Sync HttpClient.Send is not supported on browser but so is the sync-over-async
+#pragma warning disable CA1416 // 'HttpClient.Send(HttpRequestMessage, HttpCompletionOption, CancellationToken)' is unsupported on 'browser'
+                    responseMessage = Client.Send(httpRequest, HttpCompletionOption.ResponseHeadersRead, message.CancellationToken);
```
Comment (christothes) [inline sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:90]: Would a `browser` client just never call anything sync to hit this path?

Comment (pakrym) [reply sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:90]: Yeah, sync methods won't ever work.

**Review by christothes — APPROVED** [review]

File: sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs [hunk 5]
```diff
@@ -83,13 +85,16 @@ private async ValueTask ProcessAsync(HttpMessage message, bool async)
 #if NET5_0
                 if (!async)
                 {
-                    responseMessage = _client.Send(httpRequest, HttpCompletionOption.ResponseHeadersRead, message.CancellationToken);
+                    // Sync HttpClient.Send is not supported on browser but so is the sync-over-async
```
Comment (heaths) [inline sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:88]: I've read this a few times and it doesn't quite make sense. Are you saying sync-over-async is *not* supported on the browser, or it is? Perhaps reword to clarify.

File: sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs (hunk 2 above)
Comment (heaths) [reply sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:148]: Not exactly a hot path, but is this worth caching?

**Review by heaths — APPROVED** [review]

File: sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs (hunk 2 above)
Comment (pakrym) [reply sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:148]: Don't think so, it's a very non-hot path.

**Pull request merged**